sw/airborne/firmwares/rotorcraft/stabilization/stabilization_rate_indi.c

```diff
@@ -23,37 +23,140 @@
+   radio_control.values[RADIO_YAW] < -STABILIZATION_RATE_DEADBAND_R)
+
+#if PERIODIC_TELEMETRY
+#include "subsystems/datalink/telemetry.h"
+
+static void send_rate(struct transport_tx *trans, struct link_device *dev)
+{
+  pprz_msg_send_RATE_LOOP(trans, dev, AC_ID,
+                          &stabilization_rate_sp.p,
+                          &stabilization_rate_sp.q,
+                          &stabilization_rate_sp.r,
+                          &stabilization_cmd[COMMAND_THRUST]);
+}
+#endif
+
 void stabilization_rate_init(void)
 {
   stabilization_indi_init();
-}
 
+#if PERIODIC_TELEMETRY
+  register_periodic_telemetry(DefaultPeriodic, PPRZ_MSG_ID_RATE_LOOP, send_rate);
+#endif
+}
 
 void stabilization_rate_read_rc(void)
 {
-  //FIXME: make a new indi function
+  if (ROLL_RATE_DEADBAND_EXCEEDED()) {
+    stabilization_rate_sp.p = (float) radio_control.values[RADIO_ROLL] * STABILIZATION_INDI_MAX_RATE / MAX_PPRZ;
+  } else {
+    stabilization_rate_sp.p = 0;
+  }
+
+  if (PITCH_RATE_DEADBAND_EXCEEDED()) {
+    stabilization_rate_sp.q = (float) radio_control.values[RADIO_PITCH] * STABILIZATION_INDI_MAX_RATE / MAX_PPRZ;
+  } else {
+    stabilization_rate_sp.q = 0;
+  }
+
+  if (YAW_RATE_DEADBAND_EXCEEDED() && !THROTTLE_STICK_DOWN()) {
+    stabilization_rate_sp.r = (float) radio_control.values[RADIO_YAW] * STABILIZATION_INDI_MAX_RATE / MAX_PPRZ;
+  } else {
+    stabilization_rate_sp.r = 0;
+  }
 }
 
 //Read rc with roll and yaw sitcks switched if the default orientation is vertical but airplane sticks are desired
 void stabilization_rate_read_rc_switched_sticks(void)
 {
-  //FIXME: make a new indi function
+  if (ROLL_RATE_DEADBAND_EXCEEDED()) {
+    stabilization_rate_sp.r =  - (float) radio_control.values[RADIO_ROLL] * STABILIZATION_INDI_MAX_RATE / MAX_PPRZ;
+  } else {
+    stabilization_rate_sp.r = 0;
+  }
+
+  if (PITCH_RATE_DEADBAND_EXCEEDED()) {
+    stabilization_rate_sp.q = (float) radio_control.values[RADIO_PITCH] * STABILIZATION_INDI_MAX_RATE / MAX_PPRZ;
+  } else {
+    stabilization_rate_sp.q = 0;
+  }
+
+  if (YAW_RATE_DEADBAND_EXCEEDED() && !THROTTLE_STICK_DOWN()) {
+    stabilization_rate_sp.p = (float) radio_control.values[RADIO_YAW] * STABILIZATION_INDI_MAX_RATE / MAX_PPRZ;
+  } else {
+    stabilization_rate_sp.p = 0;
+  }
+}
+
+void stabilization_rate_set_setpoint_i(struct Int32Rates *pqr)
+{
+  RATES_FLOAT_OF_BFP(stabilization_rate_sp, *pqr);
 }
 
-void stabilization_rate_enter(void)
+static void stabilization_indi_rate_calc_cmd(int32_t indi_commands[], struct FloatRates rate_ref, bool in_flight)
 {
-  stabilization_indi_enter();
+  /* Get the angular acceleration references */
+  indi.angular_accel_ref.p =  indi.reference_acceleration.rate_p * (rate_ref.p - body_rates->p);
+  indi.angular_accel_ref.q =  indi.reference_acceleration.rate_q * (rate_ref.q - body_rates->q);
+  indi.angular_accel_ref.r =  indi.reference_acceleration.rate_r * (rate_ref.r - body_rates->r);
+
+  stabilization_indi_calc_cmd(indi_commands)
```

Comment:
this function seems to be static in the stabilization indi code, it should be extern